gapis/api/vulkan/replay.go

```diff
@@ -142,12 +143,30 @@ func (a API) Replay(
 	transforms = append(transforms, newDestroyResourcesAtEOS())
 	transforms = appendLogTransforms(ctx, replayType, c, transforms)
 
+	loopStart := numOfInitialCmds
+	loopEnd := api.CmdID(len(initialCmds) +len(c.Commands) -1)
 	cmdGenerator := commandGenerator.NewLinearCommandGenerator(initialCmds, c.Commands)
-	chain := transform.CreateTransformChain(ctx, cmdGenerator, transforms, out)
-	controlFlow := controlFlowGenerator.NewLinearControlFlowGenerator(chain)
-	if err := controlFlow.TransformAll(ctx); err != nil {
-		log.E(ctx, "%v Error: %v", replayType, err)
-		return err
+
+	_, _ = loopStart, loopEnd
+	switch firstRequest.(type) {
```

Comment:
There is an already switch case a few lines before that this can be a simple if check to a value(e.g. shouldBeFrameLooped etc.) instead of introducing a new one. That would also cover other cases that would affect requirement of frameLooping to be detected before here.